source/Engine.cpp

```diff
@@ -1414,7 +1414,7 @@ void Engine::CalculateStep()
 
 	// Step the weather.
 	for(Weather &weather : activeWeather)
-		weather.Step(newVisuals);
+		weather.Step(newVisuals, flagship ? flagship->Position() : center);
```

Comment:
Do we need to use the flagship position at all? `center` should be the same as the the flagship position at the start of the step, with the benefit of being ready for whenever we support viewport pan.

(Note that a "paused" viewport pan would require changing how effects are generated for system-wide hazards, since the player may pan beyond the arbitrary range of generated effects without additional calls to `Weather::Step` being made.)

Reply:
`center` gets updated later in the function, so if we simply use `center` then there'll be a delay of 1 frame to the weather center if I'm not mistaken.